### ISSUE TITLE
Em dashes should have a pause in TTS

### ISSUE THREAD

**Issue by 413Michele** [edrlab/thorium-reader#1607]
Em dashes are used in books as an alternative punctuation to commas, parentheses or even colons, as explained in [this page](https://www.thepunctuationguide.com/em-dash.html#producing). Thorium's TTS ignores the em dash though, and there is no pause in the speech.

Given that the em dash has a precise semantic meaning, I think the TTS should have a short pause when it encounters one in the text, maybe with the same duration of the comma one.

**Comment by 413Michele** [comment]
Is this something doable or it depends on the specific system TTS?

**Comment by danielweck** [comment]
Hello, thank you for bringing this to our attention.

Thorium's TTS "read aloud" feature relies on the underlying web browser technogogy known as Web Speech API, which itself delegates some of its functionality to platform-specific libraries (i.e. Windows, Mac, Linux).

Thorium is built on top of the Chromium web browser engine shipped by Electron (Chromium is also used in the Google Chrome web browser, Microsoft Edge, and quite a few others).

Chromium's implementation of Web Speech API supports SSML features which Thorium could (in principle) use to implement pause / prosody, pronunciation, etc. that are not handled by default (i.e. when just feeding plain text to the underlying TTS engine).

Thorium already implements some text fragmentation logic in order to extract reasonably-sized utterances from the document markup, for example HTML `span` elements inside `p` paragraphs generally don't produce an utterance of their own, unless they are semantically or structurally significant.

In addition to this markup-level processing, Thorium also splits long sections of unmarked text (usually, paragraphs) into shorter sentences, so that the previous/next playback navigation granularity is finer.

We could plug additional functionality in this existing processing pipeline. I think SSML is key to delivering a more sophisticated playback experience.

We are a tiny technical team with several other high development priorities so I cannot promise anything. We will report back here when progress is made.

Regards, Daniel

**Comment by 413Michele** [comment]
Of course, personally I think it's incredible that Thorium and its backend were created from scratch and are already one of the best readers on desktop PCs and a new open source content protection ecosystem (which is sadly necessary).

I'm happy to know this is technically possible, I think there are no e-book readers at the moment that deal with SSML, so this is already promising!

**Comment by formlessdao** [comment]
I personally would have liked to see focus on the speech to text as the actual priority of your project since supposedly accessibility is such a big focus. It was absolute dumb luck I found your project. I have not really seen it mention in most lists of ebook reader software. 
As it currently is the Voices in Windows 10 that I have sound good while reading the word but there is a clear unpleasant robotic rhythm while it transitions from word to word.
The playback rate increments is too significant and the increments themselves are too few and nothing under 0.5X.
I want to be able to change the pitch. 
If there is a ? or !  when a new sentence starts and has them it should be able to change intonation. for that sentence.
I can also not find how to repeat a sentence or go to the next sentence or how to go back to the previous sentence, no shortcuts for that.

**Comment by danielweck** [comment]
This is not something we can fix in Thorium, I am afraid. To be tracked with the evolution of TTS engine, Chromium web browser, AI-LLM voice / speech models.
